salt/__init__.py

```diff
@@ -131,60 +131,16 @@ def exec_module(self, module):
 
 
 def __define_global_system_encoding_variable__():
-    import sys
-
-    # This is the most trustworthy source of the system encoding, though, if
-    # salt is being imported after being daemonized, this information is lost
-    # and reset to None
-    encoding = None
-
-    if not sys.platform.startswith("win") and sys.stdin is not None:
-        # On linux we can rely on sys.stdin for the encoding since it
-        # most commonly matches the filesystem encoding. This however
-        # does not apply to windows
-        encoding = sys.stdin.encoding
-
-    if not encoding:
-        # If the system is properly configured this should return a valid
-        # encoding. MS Windows has problems with this and reports the wrong
-        # encoding
-        import locale
-
-        try:
-            encoding = locale.getencoding()
-        except AttributeError:
-            # Python < 3.11
-            encoding = locale.getpreferredencoding(do_setlocale=True)
-
-        # This is now garbage collectable
-        del locale
-
-        if not encoding:
-            # This is most likely ascii which is not the best but we were
-            # unable to find a better encoding. If this fails, we fall all
-            # the way back to ascii
-            encoding = sys.getdefaultencoding()
-        if not encoding:
-            if sys.platform.startswith("darwin"):
-                # Mac OS X uses UTF-8
-                encoding = "utf-8"
-            elif sys.platform.startswith("win"):
-                # Windows uses a configurable encoding; on Windows, Python uses the name “mbcs”
-                # to refer to whatever the currently configured encoding is.
-                encoding = "mbcs"
```

Comment:
This section of code should remain for darwin and windows.

Reply:
Starting with python 3.6 python is using `utf-8` internally and using the Unicode methods instead of the ANSI methods in Windows. As far as I can tell, mcbs only applies to Windows.

https://peps.python.org/pep-0529/

The original code was always hitting `locale.getdefaultlocale()[-1]` on Windows, which returns `cp1252`. Which was problematic because it doesn't not support Unicode and required that we pass encoding parameters everywhere. `utf-8` is a more sane default, especially since we are no longer supporting Python2.

The `sys.getdefaultencoding` function returns `utf-8` on darwin, so the behavior should be the same.

Reply:
cp1252 is correct output for English windows.  Even if python is utf-8 Windows is not utf-8 for the most part.  And this is about https://github.com/saltstack/salt/issues/65777 which is one Windows application which can read utf-8.
The correct fix is to add to each state an output encoding.

I strongly advise that this change is tested on multiple installation of Windows with different default languages.  Or even reach out to everyone who has complain about a similar issue in Windows before.

Unfortunately we can not change all the 3 party application which expect a file in the encoding of mbcs  i.e. cp1252 and all other other encoding types for all the languages Windows supports.

Reply:
> I also intend to revisit the to_mbcs functions in the registry module to see if they are still needed.
That was mostly about python 2 winapi functions only supporting mbcs,   where python 3 has a newer winapi which does not have the issue.